clients/src/main/java/org/apache/kafka/common/requests/RequestHeader.java

```diff
@@ -118,7 +118,6 @@ public static RequestHeader parse(ByteBuffer buffer) {
             // We derive the header version from the request api version, so we read that first.
             // The request api version is part of `RequestHeaderData`, so we reset the buffer position after the read.
             int position = buffer.position();
-            int requestHeaderSize = buffer.remaining();
```

Comment:
This is not correct. The buffer in the payload is actually [header + message]. So, if we do `buffer.remaining()` here, we'll get the wrong header size. Fix it by removing the cache setting here.